src/cursor/abstract_cursor.ts

```diff
@@ -825,6 +833,11 @@ export abstract class AbstractCursor<
     this.isKilled = false;
     this.initialized = false;
 
+    this.cursorClient.s.activeCursors.add(this);
+    if (!this.listeners('close').includes(removeActiveCursor)) {
+      this.once('close', removeActiveCursor);
+    }
```

Comment:
Any chance that this and the constructor should share more code? This change here feels like something that could have been really easy to miss if you didn't happen to think of it :)

Reply:
Yea "setupTracking" helper of sorts seems worth having, will do, ty